### ISSUE TITLE
remove if the if condition by assigning function statically

### ISSUE THREAD

**Issue by KrishnaPG** [jchook/uuid-random#5]
The `crypto` availability is being checked every time `randomBytes` method is called [here](https://github.com/jchook/uuid-random/blob/master/index.js#L53)

This if else condition checking can be done just once at the beginning, and assign the correct function to a variable, something like below: 
````
const randomBytes = (typeof crypto !== 'undefined') ? randomCrypto : randomMath;

let bufIdx = uuid.BUFFER_SIZE;
function randomCrypto(n) {
      if ((++bufIdx + n) > uuid.BUFFER_SIZE) {
        bufIdx = 0;
        if (crypto.getRandomValues) {
          buf = new Uint8Array(uuid.BUFFER_SIZE);
          crypto.getRandomValues(bytes);
        } else if (crypto.randomBytes) {
          buf = crypto.randomBytes(uuid.BUFFER_SIZE);
        } else {
          throw new Error('Non-standard crypto library');
        }
      }
      return buf.slice(bufIdx, bufIdx + n);
}
function randomMath(n) { 
      r = [];
      for (i = 0; i < n; i++) {
        r.push(getRandomInt(0, 255));
      }
      return r;
}
````
This further improves the performance.

**Comment by jchook** [comment]
Hey thanks very much for your thoughtful suggestion.

I tried the changes and didn't see any material performance gains. However, in testing, I did find a bug! So ultimately I think it was a big win.

**Comment by KrishnaPG** [comment]
The performance gains will not be visible at the code level. They will be at the energy consumption level of the underlying CPUs, as the if-else branching logic affects the Processor cache pipeline, especially when the processors have branch-prediction logic built-in. If you want to measure the performance at the code level, then try with RDTSC performance counters.

**Comment by jchook** [comment]
Interesting.

What do you think of this PR? #6

**Comment by KrishnaPG** [comment]
Looks fine to me, except for one small thing:

> At Line 76, the else condition for `randomBytesMath` may need to be outside the if (typeof crypto !== 'undefined'), since it should get executed even when crypto === undefined. Currently it looks like it will not run if crypto is undefined.

**Comment by KrishnaPG** [comment]
Also, It would be great if `nacl` is supported, if possible. 
Ref: [Best Practices of Security](https://gist.github.com/atoponce/07d8d4c833873be2f68c34f9afc5a78a)

[TweetNACL](https://github.com/dchest/tweetnacl-js) 

`nacl.randomBytes` internally picks up the right method based on the current platform.

**Comment by jchook** [comment]
I got around to implementing this performance fix in #11 and also exposed `uuid.randomBytes()` so you can override it with TweetNACL. Thanks again for the great suggestions.
